code/modules/atmospherics/machinery/components/turbine/turbine.dm

```diff
@@ -0,0 +1,372 @@
+	SetInitDirections()
+	var/obj/machinery/atmospherics/node = nodes[1]
+	if(node)
+		if(src in node.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
+			node.disconnect(src)
+		nodes[1] = null
+	if(parents[1])
+		nullifyPipenet(parents[1])
+
+	atmosinit()
+	node = nodes[1]
+	if(node)
+		node.atmosinit()
+		node.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
+
+/obj/machinery/atmospherics/components/unary/turbine/turbine_inlet/update_appearance()
+	. = ..()
+	if(connected)
+		icon_state = "[base_icon]_conn"
+	else
+		icon_state = base_icon
+	if(on)
+		icon_state = "[base_icon]_on"
+
+/obj/machinery/atmospherics/components/unary/turbine/turbine_inlet
+	icon_state = "turbine_inlet"
+	name = "turbine inlet port"
+	desc = "Input port for the turbine."
+	circuit = /obj/item/circuitboard/machine/turbine_inlet
+	initial_volume = 500
+	base_icon = "turbine_inlet"
+
+/obj/machinery/atmospherics/components/unary/turbine/turbine_outlet
+	icon_state = "turbine_outlet"
+	name = "turbine outlet port"
+	desc = "Output port for the turbine."
+	circuit = /obj/item/circuitboard/machine/turbine_outlet
+	initial_volume = 1000
+	base_icon = "turbine_outlet"
+
+/obj/machinery/power/turbine
+	icon = 'icons/obj/atmospherics/components/turbine.dmi'
+	layer = OBJ_LAYER
+	///Check to see if we are connected to the other parts
+	var/connected = FALSE
+	///base icon for appearance changes
+	var/base_icon = ""
+	///Check if the machine is turned on
+	var/on = FALSE
+
+/obj/machinery/power/turbine/update_appearance()
+	. = ..()
+	if(connected)
+		icon_state = "[base_icon]_conn"
+	else
+		icon_state = base_icon
+	if(on)
+		icon_state = "[base_icon]_on"
+
+/obj/machinery/power/turbine/attackby(obj/item/I, mob/user, params)
+	if(!connected)
+		if(default_deconstruction_screwdriver(user, "[base_icon]_open", "[base_icon]", I))
+			update_appearance()
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/power/turbine/turbine_shaft
+	icon_state = "turbine_shaft"
+	name = "turbine shaft"
+	desc = "Main body of the turbine multipart."
+	circuit = /obj/item/circuitboard/machine/turbine_shaft
+	base_icon = "turbine_shaft"
+	density = TRUE
+
+/obj/machinery/power/turbine/turbine_controller
+	icon_state = "turbine_interface"
+	name = "turbine controller"
+	desc = "Controller that connects the turbine parts in one and allows control of the machine."
+	circuit = /obj/item/circuitboard/machine/turbine_controller
+	base_icon = "turbine_interface"
+	///Stores all connected parts
+	var/list/turbine_parts = list()
+	///Stores a reference to the turbine inlet
+	var/obj/machinery/atmospherics/components/unary/turbine/turbine_inlet/inlet
+	///Stores a reference to the turbine outlet
+	var/obj/machinery/atmospherics/components/unary/turbine/turbine_outlet/outlet
+	///Stores a reference to the turbine shaft
+	var/obj/machinery/power/turbine/turbine_shaft/shaft
```

Comment:
Will these cause hard deletes?

Reply:
they shouldn't because they are nulled on Destroy() of the controller and they are also nulled if any of the parts get deleted for any reason (hooked with signals).
If there is anything that i missed lemme know!